app/src/main/java/com/quran/labs/androidquran/ui/translation/TranslationView.java

```diff
@@ -68,6 +76,46 @@ public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
     ayahToolBar.setVisibility(View.GONE);
     addView(ayahToolBar, LayoutParams.WRAP_CONTENT,
         context.getResources().getDimensionPixelSize(R.dimen.toolbar_total_height));
+
+    ImageButton copyButton = (ImageButton) ayahToolBar.findViewById(R.id.cab_copy_ayah_text_menu);
+    PopupMenu popup = new PopupMenu(context,copyButton);
+
+    copyButton.setOnClickListener(new View.OnClickListener() {
+      @Override
+
+      public void onClick(View v) {
+        PopupMenu popmenu;
```

Comment:
combine lines 87 and 88.

Reply:
should we short circuit when there's only one translation and automagically copy instead of showing the menu?

Reply:
the original old copy button is still there. it is used to copy all translations. so if one translation there the old copy button does it. so i think it is not needed? what is your opinion?(the old copy button also gets the arabic text too by the way)